File: 0912-sort-an-array/0912-sort-an-array.py
class Solution:
    def sortArray(self, nums: List[int]) -> List[int]:
        min_val = -1
        max_val = -1
        count_map = {}
        for n in nums:
            if n < min_val:
                min_val = n
            elif n > max_val:
                max_val = n
            count_map[n] = count_map.get(n,0) + 1
        arr = []
        for n in range(min_val, max_val+1):
            if n in count_map:
                while count_map[n] != 0:
                    arr.append(n)
                    count_map[n] -= 1
        return arr

File: 0912-sort-an-array/0912-sort-an-array.java
class Solution {
    public int[] sortArray(int[] nums) {
        int minVal = 0;
        int maxVal = 0;
        HashMap<Integer, Integer> countMap = new HashMap<>();
        for(int n : nums) {
            if(minVal > n) {
                minVal = n;
            } else if(maxVal < n) {
                maxVal = n;
            }
            countMap.put(n, countMap.getOrDefault(n,0)+1);
        }
        int i = 0;
        while(minVal <= maxVal) {
            if(countMap.containsKey(minVal)) {
                int times = countMap.get(minVal);
                while(times != 0) {
                    nums[i] = minVal;
                    i++;
                    times--;
                }
            }
            minVal++;
        }
        return nums;
    }
}